### PR TITLE
feature/register

### PR DESCRIPTION
**Pull request by maxence-lefebvre**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
     "app-module-path": "^2.2.0",
     "classnames": "^2.2.5",
     "fast-memoize": "^2.3.2",
+    "font-awesome": "^4.7.0",
```
Comment (gautier-lefebvre) [inline package.json:11]: not fontawesome 5?

Comment (gautier-lefebvre) [reply package.json:11]: Ok my bad, it's for the reapop theme.

Comment (gautier-lefebvre) [reply package.json:11]: Resolved.

File: src/renderer/redux/app/reducers.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { combineReducers } from 'redux';
+import { ACTIONS } from './actions';
+
+export default combineReducers({
+  storeFilePath: (state = '', action) => {
```
Comment (gautier-lefebvre) [inline src/renderer/redux/app/reducers.js]: Why do you keep the storeFilePath in the store ?

File: package.json (hunk 1 above)
Comment (sylvainar) [reply package.json:11]: Do we tree-shake it ?

File: src/renderer/react/AppWrapper.jsx [hunk 3]
```diff
@@ -3,29 +3,30 @@ import PropTypes from 'prop-types';
...
 import { IntlProvider } from 'react-intl-redux';
 import { BrowserRouter as Router } from 'react-router-dom';
+import NotificationsSystem from 'reapop';
 
-export const storeShape = {
-  subscribe: PropTypes.func.isRequired,
-  dispatch: PropTypes.func.isRequired,
-  getState: PropTypes.func.isRequired,
-};
+import { reapopThemeShape, storeShape } from './app-wrapper.shapes';
```
Comment (sylvainar) [inline src/renderer/react/AppWrapper.jsx]: Where do we put our shapes ? With the component or in a `react/shapes` folder ?

Comment (maxence-lefebvre) [reply src/renderer/react/AppWrapper.jsx]: I dunno

File: src/renderer/redux/app/reducers.js (hunk 2 above)
Comment (maxence-lefebvre) [reply src/renderer/redux/app/reducers.js]: To display it to the user, to tell him to do some backups of it

File: package.json (hunk 1 above)
Comment (maxence-lefebvre) [reply package.json:11]: I hope I will be able to kick it out of the project once we build our first theme (waiting for some designs to drop out of a truck)

File: src/renderer/react/app/login/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+import messages from './login.messages';
+
+const mapDispatchToProps = dispatch => ({
+  actions: bindActionCreators({
+    login: loginAction,
+    notify,
+  }, dispatch),
+});
+
+export const LoginContainer = (Presentation) => {
```
Comment (sylvainar) [inline src/renderer/react/app/login/Login.jsx:21]: What's the point of writing an HOC here ?

Comment (maxence-lefebvre) [reply src/renderer/react/app/login/Login.jsx:21]: Is it not a best practice ?

File: src/main/state/encrypt/index.js [hunk 5]
```diff
@@ -4,6 +4,17 @@ const crypto = require('crypto');
...
 
+/**
+ * Is user registered ?
+ * That would mean that the app is initialized and that the store file exists.
+ *
+ * @param {string} storeFilePath - Where to find the store file.
+ * @returns {bool} Whether the store file exists.
+ */
+function isInitialized(storeFilePath) {
+  return fse.pathExists(storeFilePath);
```
Comment (gautier-lefebvre) [inline src/main/state/encrypt/index.js:16]: Isn't it async ?

File: src/renderer/react/app/login/Login.jsx (hunk 4 above)
Comment (sylvainar) [reply src/renderer/react/app/login/Login.jsx:21]: I think it's a best practice when the code from the HOC will be re-used, or else it just adds useless complexity (but it's my POV)

File: src/main/state/encrypt/index.js (hunk 5 above)
Comment (maxence-lefebvre) [reply src/main/state/encrypt/index.js:16]: Resolved.

File: src/renderer/react/app/login/Login.jsx (hunk 4 above)
Comment (maxence-lefebvre) [reply src/renderer/react/app/login/Login.jsx:21]: Well the thing is, I believe it will be easier to unit test both parts this way :D

Comment (gautier-lefebvre) [reply src/renderer/react/app/login/Login.jsx:21]: @sylvainar It's easier to think of your container as something completely isolated from the presentation when you write them as HOC with the presentation as parameter.

File: src/renderer/services/state/index.js [hunk 6]
```diff
@@ -1,5 +1,25 @@
...
+ * Meaning, is the user locally registered ?
+ *
+ * @returns {Promise<bool>} - Is the app initialized ?
+ */
+export const isInitialized = () => send(IS_INITIALIZED);
+
+/**
+ * Gets the store file path used by developer-vault.
+ *
+ * @returns {Promise<string>} - The store file path.
```
Comment (gautier-lefebvre) [inline src/renderer/services/state/index.js:20]: I think we should just specify the 'string' type in `@returns`, and setting `@async` to know that this is a Promise.

**Review by gautier-lefebvre — APPROVED** [review]

**Pull request merged**